### PR TITLE
Move old business viewpoint id to a proper field

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

File: terra_opp/models.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ class Viewpoint(BaseLabelModel):
     )
     properties = JSONField(_("Properties"), default=dict, blank=True)
     related = GenericRelation("datastore.RelatedDocument")
+    viewpoint_number = models.PositiveIntegerField(_("Viewpoint number"), unique=True)
```
Comment (jrmi) [inline terra_opp/models.py:66]: This is the viewpoint model so I think we don't need the "viewpoint prefix". Also number is too generic so may be something like `identifier` ?

Comment (CFarcy) [reply terra_opp/models.py]: Done.

**Review by jrmi — APPROVED** [review]

File: terra_opp/tests/factories.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ class ViewpointFactory(factory.django.DjangoModelFactory):
     )
     city = factory.SubFactory("terra_opp.tests.factories.CityFactory")
     themes = factory.RelatedFactory("terra_opp.tests.factories.ThemeFactory")
+    identifier = factory.Sequence(lambda n: n)
```
Comment (jrmi) [inline terra_opp/tests/factories.py:23]: I think it should be the pk also, to use it in url

**Review by jrmi — CHANGES_REQUESTED** [review]
Need to change the url

File: terra_opp/models.py [hunk 3]
```diff
@@ -98,6 +100,17 @@ class Meta:
...
 
+    def save(self, *args, **kwargs):
+        if self.pk is None and not self.identifier:
+            max_identifier = (
+                Viewpoint.objects.all()
+                .aggregate(Max("identifier"))
+                .get("identifier__max")
+            )
+            # aggregate return None if no value exists, i.e empty queryset
+            self.identifier = max_identifier + 1 if max_identifier is not None else 0
```
Comment (jrmi) [inline terra_opp/models.py:111]:
```suggestion
            self.identifier = max_identifier + 1 if max_identifier is not None else 1
```

Comment (CFarcy) [reply terra_opp/models.py:111]: identifier should start at `1`?

**Review by jrmi — APPROVED** [review]

**Pull request merged**